src/hooks.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import type {Config} from 'cli-engine-config'
+import path from 'path'
+
+export default class Hooks {
+  config: Config
+  hooks: {[name: string]: string[]}
+
+  constructor ({config}: {config: Config}) {
+    this.config = config
+    this.hooks = {}
+    let cliEngine: any = config.pjson['cli-engine'] || {}
+    for (let [name, script] of Object.entries(cliEngine['hooks'] || {})) {
+      if (typeof script !== 'string') throw new Error(`expected ${name} to be a string`)
```

Comment:
I would have to make the type mixed though which isn't ideal. this conditional is really just to make flow happy